slobrok/src/vespa/slobrok/server/local_rpc_monitor_map.cpp

```diff
@@ -97,95 +100,94 @@ void LocalRpcMonitorMap::doAdd(const ServiceMapping &mapping) {
     auto old = _map.find(mapping.name);
     if (old != _map.end()) {
         PerService & exists = old->second;
-        if (exists.spec() == mapping.spec) {
+        if (exists.spec == mapping.spec) {
             LOG(debug, "added mapping %s->%s was already present",
                 mapping.name.c_str(), mapping.spec.c_str());
             exists.localOnly = false;
             return;
         }
         PerService removed = std::move(exists);
+        ServiceMapping toRemove{old->first, removed.spec};
```

Comment:
consider using a helper function that removes something from the map, calls stop and returns a PerService